lib/puppet/indirector/file_metadata/http.rb

```diff
@@ -0,0 +1,27 @@
+require 'puppet/file_serving/http_metadata'
+require 'puppet/indirector/file_metadata'
+require 'net/http'
+require 'puppet/network/http_pool'
+
+class Puppet::Indirector::FileMetadata::Http < Puppet::Indirector::Plain
+  desc "Retrieve file metadata from a remote HTTP server."
+
+  def find(request)
+    uri = URI( request.to_s.sub(%r{^/file_metadata/url=},'') )
+
+    use_ssl = uri.scheme == 'https'
+    connection = Puppet::Network::HttpPool.http_instance(uri.host, uri.port, use_ssl)
+
+    response = connection.head(uri.path)
```

Comment:
Here's the code snippet we were talking about so that puppet will trust additional cacerts: https://github.com/puppetlabs/puppet/blob/master/lib/puppet/forge/repository.rb#L136-L143